### ISSUE TITLE
Build failure with MinGW-w64

### ISSUE THREAD

**Issue by kleisauke** [ImageMagick/ImageMagick6#67]
I'm unable to cross-compile ImageMagick 6.9.10-80 with MinGW-w64. Here's the salient part of the build log: https://gist.github.com/kleisauke/d385c6b02d31690fb9b22a6ddecad612

Note that the multiple definition linker error also affects other dependent packages: Imagick/imagick#311

This patch fixes it for me:
```diff
diff --git a/magick/memory_.h b/magick/memory_.h
index 1111111..2222222 100644
--- a/magick/memory_.h
+++ b/magick/memory_.h
@@ -68,7 +68,7 @@ extern MagickExport void
   SetMagickMemoryMethods(AcquireMemoryHandler,ResizeMemoryHandler,
     DestroyMemoryHandler);
 
-inline MagickExport MagickBooleanType HeapOverflowSanityCheck(
+inline MagickBooleanType HeapOverflowSanityCheck(
   const size_t count,const size_t quantum)
 {
   if ((count == 0) || (quantum == 0))
@@ -81,7 +81,7 @@ inline MagickExport MagickBooleanType HeapOverflowSanityCheck(
   return(MagickFalse);
 }
 
-inline MagickExport MagickBooleanType HeapOverflowSanityCheckGetSize(
+inline MagickBooleanType HeapOverflowSanityCheckGetSize(
   const size_t count,const size_t quantum,size_t *const extent)
 {
   size_t
diff --git a/magick/magick-config.h b/magick/magick-config.h
index 1111111..2222222 100644
--- a/magick/magick-config.h
+++ b/magick/magick-config.h
@@ -162,7 +162,7 @@ extern "C" {
 #  define __has_builtin(x) 0
 #endif
 
-#if __STDC_VERSION__ > 201112L
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ > 201112L && !defined(__MINGW32__)
 # define MAGICKCORE_HAVE_STDC_ALIGNED_ALLOC 1
 #endif
 
````

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by kleisauke** [comment]
I confirm that this build failure has been resolved in ImageMagick 6.9.10-82. Thanks for fixing this! I'll close this issue.
